pkg/policies/codeowners/codeowners.go

```diff
@@ -0,0 +1,240 @@
+	// OptConfig is the standard org-level opt in/out config, RepoOverride applies to all
+	// BP config.
+	OptConfig config.OrgOptConfig `json:"optConfig"`
+
+	// Action defines which action to take, default log, other: issue...
+	Action string `json:"action"`
+
+	//TODO add default contents for "fix" action
+}
+
+// RepoConfig is the repo-level config for Branch Protection
+type RepoConfig struct {
+	// OptConfig is the standard repo-level opt in/out config.
+	OptConfig config.RepoOptConfig `json:"optConfig"`
+
+	// Action overrides the same setting in org-level, only if present.
+	Action *string `json:"action"`
+}
+
+type repositories interface {
+	GetCodeownersErrors(ctx context.Context, owner, repo string) (*github.CodeownersErrors, *github.Response, error)
+}
+
+type mergedConfig struct {
+	Action string
+}
+
+type details struct {
+	Enabled         bool
+	URL             string
+	CodeownersFound bool
+	ErrorCount      int
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, config.ConfigLevel, interface{}) error
+
+var configIsEnabled func(ctx context.Context, o config.OrgOptConfig, orc, r config.RepoOptConfig, c *github.Client, owner, repo string) (bool, error)
+
+func init() {
+	configFetchConfig = config.FetchConfig
+	configIsEnabled = config.IsEnabled
+}
+
+// Codeowners is the CODEOWNERS policy object, implements policydef.Policy.
+type Codeowners bool
+
+// NewCodeowners returns a new CODEOWNERS policy.
+func NewCodeowners() policydef.Policy {
+	var s Codeowners
+	return s
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (s Codeowners) Name() string {
+	return polName
+}
+
+// Check performs the policy check for CODEOWNERS policy based on the
+// configuration stored in the org/repo, implementing policydef.Policy.Check()
+func (s Codeowners) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	return check(ctx, c.Repositories, c, owner, repo)
+}
+
+// Check whether this policy is enabled or not
+func (s Codeowners) IsEnabled(ctx context.Context, c *github.Client, owner, repo string) (bool, error) {
+	oc, orc, rc := getConfig(ctx, c, owner, repo)
+	return configIsEnabled(ctx, oc.OptConfig, orc.OptConfig, rc.OptConfig, c, owner, repo)
+}
+
+func check(ctx context.Context, rep repositories, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	oc, orc, rc := getConfig(ctx, c, owner, repo)
+	enabled, err := configIsEnabled(ctx, oc.OptConfig, orc.OptConfig, rc.OptConfig, c, owner, repo)
+	var d details
+	d.Enabled = enabled
+	d.URL = ""
+	if err != nil {
+		return nil, err
+	}
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+
+	codeownererrors, _, err := rep.GetCodeownersErrors(context.Background(), owner, repo)
+
+	// this is a 404, CODEOWNERS is not the in the repository
+	if err != nil {
```

Comment:
You need to check the response here. It could be a 404, or another error
example: https://github.com/ossf/allstar/blob/9fd9b106f0cd1f3754c461d9c123d39a89024f6e/pkg/config/location.go#L91-L108

Reply:
Yeah it's a weird pattern, from what I can see when debugging, both the response and the `err` have the 404 information, because the error has the `github.Response` inside of it
Maybe I could handle different cases like a 500, and not create a Github issue at all in that case